src/coreclr/vm/jitinterface.cpp

```diff
@@ -11777,13 +11777,13 @@ bool CEEInfo::getStaticFieldContent(CORINFO_FIELD_HANDLE fieldHnd, uint8_t* buff
     {
         if (field->IsObjRef())
         {
-            GCX_COOP();
-
-            _ASSERT(!field->IsRVA());
-            _ASSERT(valueOffset == 0); // there is no point in returning a chunk of a gc handle
-            _ASSERT((UINT)bufferSize == field->GetSize());
+            // there is no point in returning a chunk of a gc handle
```

Comment:
Is the same change needed in the AOT/R2R getStaticFieldContent?

https://github.com/dotnet/runtime/blob/8c68710084b861c44b6868c0bd972ce24aed55dd/src/coreclr/tools/aot/ILCompiler.RyuJit/JitInterface/CorInfoImpl.RyuJit.cs#L2311

Reply:
I think @EgorBo or @MichalStrehovsky would have to answer that. The following asserts look potentially the same as in the jitinterface and would need to be converted to allow "garbage" input:

https://github.com/dotnet/runtime/blame/8c68710084b861c44b6868c0bd972ce24aed55dd/src/coreclr/tools/aot/ILCompiler.RyuJit/JitInterface/CorInfoImpl.RyuJit.cs#L2360-L2361

Reply:
Yeah I think these asserts are redundant on NAOT side as well since we should be robust to potentially invalid IL that tries to do something illegal